services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerService.java

```diff
@@ -120,17 +125,45 @@ public VeniceControllerService(
     }
     // The admin consumer needs to use VeniceHelixAdmin to update Zookeeper directly
     consumerServicesByClusters = new HashMap<>(multiClusterConfigs.getClusters().size());
-    /** N.B. The code below is copied from {@link com.linkedin.venice.controller.init.SystemSchemaInitializationRoutine */
-    // BiConsumer<Integer, Schema> newSchemaEncountered = (schemaId, schema) -> internalAdmin.addValueSchema(
-    // "?", // TODO: Figure out a clean way to retrieve the cluster name param
-    // AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getSystemStoreName(),
-    // schema.toString(),
-    // schemaId,
-    // DirectionalSchemaCompatibilityType.NONE,
-    // false);
-    KafkaValueSerializer kafkaValueSerializer = new OptimizedKafkaValueSerializer(
-    // newSchemaEncountered // TODO: Wire in this hook once we figure out a clean way to do it
-    );
+
+    /**
+     * Register a callback function to handle the case when a new KME value schema is encountered when the child controller
+     * consumes the admin topics.
+     */
+    BiConsumer<Integer, Schema> newSchemaEncountered = (schemaId, schema) -> {
+      LOGGER.info("Encountered a new KME value schema (id = {}), proceed to register", schemaId);
+      String systemClusterName = multiClusterConfigs.getSystemSchemaClusterName();
+      VeniceControllerConfig systemStoreClusterConfig = multiClusterConfigs.getControllerConfig(systemClusterName);
+      try {
+        ControllerClientBackedSystemSchemaInitializer schemaInitializer =
```

Comment:
Can we maybe make this a member variable?  Might make testing easier.

Reply:
Do you mean `newSchemaEncountered` ? yeah, I think I can make it a class member.